src/js/components/Button/Button.js

```diff
@@ -84,17 +84,30 @@ class Button extends Component {
     } = this.props;
     const { hover } = this.state;
 
-    let buttonIcon = icon;
     // only change color if user did not specify the color themselves...
-    if (primary && icon && !icon.props.color) {
+    let buttonIcon = icon;
+    // nonprimary button icon styling with custom theme
+    if (!primary && icon && !icon.props.color) {
```

Comment:
It might be a bit clearer to wrap all of this inside an `if (icon && !icon.props.color)` and then have an if/else based on whether `primary` is set or not. Also, rather than repeat the `cloneElement()` multiple times, just have the if/else stuff set an `iconColor` property and then have one `cloneElement()` at the end.